packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -1072,7 +1086,10 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
             this.sessionPool.setMaxListeners(20);
         }
 
+        await this._initializeRequestProviders();
         await this._loadHandledRequestCount();
+
+        this.events.on(EventType.MIGRATING, this._pauseOnMigration);
```

Comment:
This event listener is attached later in the `run` implementation. 

https://github.com/apify/crawlee/blob/99af95e0dda511718b45cd41452589260c69909a/packages/basic-crawler/src/internals/basic-crawler.ts#L935

What is the reason to do it here?